### PR TITLE
Remove scheduler

### LINKED ISSUE #412: Remove pd scheduler in backup
### Is your feature request related to a problem? Please describe:

Add an option that BR can remove pd leader related scheduler automatically. just like restore [does](https://github.com/pingcap/br/blob/ef923324feec7cc8d58f7818cc7b6f8758cc4a0d/pkg/task/restore.go#L359-L423)
In our test: 
with balance leader scheduler cost:  9m33s
without balance leader scheduler cost: 2m31s

this could happen because there are so many incomplete retry when leader transfer.

### Describe the feature you'd like:

Add an option that can remove scheduler automatically to speed up backup.

### Describe alternatives you've considered:

This change has some impact on backup cluster.so we can set default value to false.

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve? 
fixed #412.

### What is changed and how it works?
Firstly, we extract the logic of removing schedulers to `conn` package, as a method of `conn.Manager`.
Then, we add a flag that controls whether to remove scheduler when backing up.
Finally, we call the extracted method on backup if the flag is set. 

### Check List 

Tests 

 - Integration test

### Release Note

 - Add an option that can be used to speed up backing up by changing some cluster config.

### CODE REVIEW

File: pkg/conn/scheduler_utils.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		}
+	}
+	scheduler, err := removePDLeaderScheduler(ctx, mgr, needRemoveSchedulers)
+	if err != nil {
+		return utils.Nop, nil
+	}
+
+	stores, err := mgr.GetPDClient().GetAllStores(ctx)
+	if err != nil {
+		return utils.Nop, err
```
Comment (kennytm) [inline pkg/conn/scheduler_utils.go:107]: i think you shouldn't return Nop given the removePDLeaderScheduler has succeeded.

File: pkg/task/backup.go [hunk 2]
```diff
@@ -64,6 +66,9 @@ func DefineBackupFlags(flags *pflag.FlagSet) {
 	flags.Int64(flagGCTTL, backup.DefaultBRGCSafePointTTL, "the TTL (in seconds) that PD holds for BR's GC safepoint")
 	flags.String(flagCompressionType, "zstd",
 		"backup sst file compression algorithm, value can be one of 'lz4|zstd|snappy'")
+
+	flags.Bool(flagRemoveSchedulers, false,
+		"remove some of PD schedulers to speed up backup, but will make influence to cluster")
```
Comment (kennytm) [inline pkg/task/backup.go:71]:
```suggestion
		"disable the balance, shuffle and region-merge schedulers in PD to speed up backup")
```

File: pkg/task/backup_raw.go [hunk 3]
```diff
@@ -45,6 +48,8 @@ func DefineRawBackupFlags(command *cobra.Command) {
 	command.Flags().StringP(flagEndKey, "", "", "backup raw kv end key, key is exclusive")
 	command.Flags().String(flagCompressionType, "zstd",
 		"backup sst file compression algorithm, value can be one of 'lz4|zstd|snappy'")
+	command.Flags().Bool(flagRemoveSchedulers, false,
+		"remove some of PD schedulers to speed up backup, but will make influence to cluster")
```
Comment (kennytm) [inline pkg/task/backup_raw.go:52]: (ditto)

File: tests/br_other/run.sh [hunk 4]
```diff
@@ -72,9 +73,11 @@ sleep 1
 curl "http://localhost:$PPROF_PORT/debug/pprof/trace?seconds=1" 2>&1 > /dev/null
 echo "pprof started..."
 
+curl http://$PD_ADDR/pd/api/v1/config/schedule | grep '"disable": true'
+
 backup_fail=0
 echo "another backup start expect to fail due to last backup add a lockfile"
-run_br --pd $PD_ADDR backup full -s "local://$TEST_DIR/$DB/lock" --concurrency 4 || backup_fail=1
+run_br --pd $PD_ADDR backup full -s "local://$TEST_DIR/$DB/lock"  --concurrency 4 || backup_fail=1
```
Comment (3pointer) [inline tests/br_other/run.sh:80]: remove space

File: pkg/task/backup.go [hunk 5]
```diff
@@ -64,6 +66,9 @@ func DefineBackupFlags(flags *pflag.FlagSet) {
 	flags.Int64(flagGCTTL, backup.DefaultBRGCSafePointTTL, "the TTL (in seconds) that PD holds for BR's GC safepoint")
 	flags.String(flagCompressionType, "zstd",
 		"backup sst file compression algorithm, value can be one of 'lz4|zstd|snappy'")
+
+	flags.Bool(flagRemoveSchedulers, false,
+		"disable the balance, shuffle and region-merge schedulers in PD to speed up backup")
```
Comment (3pointer) [inline pkg/task/backup.go:71]: maybe we should hide it or let user know this will impact on online cluster.

**Pull request merged**